src/Stratis.SmartContracts.Executor.Reflection/ReflectionVirtualMachine.cs

```diff
@@ -18,48 +23,86 @@ public class ReflectionVirtualMachine : ISmartContractVirtualMachine
     {
         private readonly InternalTransactionExecutorFactory internalTransactionExecutorFactory;
         private readonly ILogger logger;
+        private readonly Network network;
+        private readonly ISmartContractValidator validator;
         public static int VmVersion = 1;
 
-        public ReflectionVirtualMachine(
+        public ReflectionVirtualMachine(ISmartContractValidator validator,
             InternalTransactionExecutorFactory internalTransactionExecutorFactory,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            Network network)
         {
+            this.validator = validator;
             this.internalTransactionExecutorFactory = internalTransactionExecutorFactory;
             this.logger = loggerFactory.CreateLogger(this.GetType());
+            this.network = network;
         }
 
         /// <summary>
         /// Creates a new instance of a smart contract by invoking the contract's constructor
         /// </summary>
-        public VmExecutionResult Create(byte[] contractCode,
-            ISmartContractExecutionContext context,
+        public VmExecutionResult Create(
```

Comment:
The call and create are very "heavy" now Rowam, do we intend to refactor to make them easier to read? A bit SRP there dude;)

Reply:
Yes, I have more plans for these :)